### DISCUSSION TITLE
Display items that are done and archived

### DISCUSSION THREAD

**Discussion by pmiddend** [alphapapa/org-ql#458] [Q&A] [answered]
I'm frequently marking items as `DONE` and then immediately archive them. Is it possible to search for these items (in an `org-ql-search` expression) and display them (at least if they're done and archived today).

**Comment by alphapapa** [comment]
I guess it depends on your archiving configuration.

**Comment by kaibagley** [comment]
I'd like to know this too, however in my case I archive entries older than 2 months into another file which is GPG encrypted (todo.org.gpg_archive.gpg). Not sure if there is, but is there a way to have these be included in the search? If they aren't open, there would be a pin entry for decrypting them, then search as normal

**Reply by alphapapa** [reply]
`org-ql` will search whatever files you tell it to search.  If you want to search in archive files, tell `org-ql` to search in those files.

**Reply by kaibagley** [reply]
Thanks for the reply. Could you please give me a hint on how I can perhaps add an option to the interactive query (alongside "all", "org-directory", etc.), such as "archive-files", which would tell `org-ql` to run the query only on those?

**Reply by alphapapa** [reply]
It would be easier if I had a better idea of how much you already know about Emacs, Emacs Lisp, etc.  IOW, I'm not sure how much code you're asking me to write.  How much of org-ql's code have you looked at yet?

**Reply by kaibagley** [reply]
I've been using Emacs for about a year, but have done little to no elisp outside of simple init.el stuff.

I can see here: https://github.com/alphapapa/org-ql/blob/fcb4e3ee62d2c6f9d2a9a8a510a50dd1a6eceb5e/org-ql-search.el#L186 that `BUFFERS-FILES` is passed the output of `org-ql-view--complete-buffers-files`, which is defined here: https://github.com/alphapapa/org-ql/blob/fcb4e3ee62d2c6f9d2a9a8a510a50dd1a6eceb5e/org-ql-view.el#L1086, I can try altering this:

```
(setq org-archive-files '("file1.org_archive", "file2.org_archive"))
(defun org-ql-view--complete-buffers-files ()
  "Return value for `org-ql-view-buffers-files' using completion."
  (cl-labels ((initial-input ()
                (when org-ql-view-buffers-files
                  (org-ql-view--contract-buffers-files
                   org-ql-view-buffers-files))))
    (if (and org-ql-view-buffers-files
             (bufferp org-ql-view-buffers-files))
        ;; Buffers can't be input by name, so if the default value is a buffer, just use it.
        ;; TODO: Find a way to fix this.
        org-ql-view-buffers-files
      (org-ql-view--expand-buffers-files
       (completing-read "Buffers/Files: "
                        (list 'buffer 'org-agenda-files 'org-directory 'all 'org-archive-files)
                        nil nil (initial-input))))))
```

And then altering a lot more code, but I don't want to mess around with too much. I thought I'd ask in case there was some much easier way of doing it.

**Reply by alphapapa** [reply]
It depends on what exactly you're wanting to do.  If you only ever want to search the same two files, then you could just write your own function that calls `org-ql-search` with those files.  If you want to select the archive files each time, you could write a function that selects the files first (e.g. with completion, or `directory-files-recursively`), and then passes them to `org-ql-search`.

If you really want to add an option to the org-ql completion like that, then you'll have to account for it in both the "expand" and "contract" functions, probably.  The "hard" part is in the "contraction," because that function takes a value and tries to determine which compact representation applies, if any.

Generally I'd recommend not changing the completion-related code for just your own sake.  (If you wanted to propose a general improvement to it, that would be different.)  It would be easier and more maintainable to just write your own function to call org-ql functions with the arguments you need.

**Reply by kaibagley** [reply, marked as answer]
Yeah that's very true, my own function which calls `org-ql-search` specifically for those files is definitely the way to go.

Thanks for taking the time to explain this to me, you're a legend!
